lib/src/authentication/client.dart

```diff
@@ -129,26 +129,29 @@ Future<T> withAuthenticatedClient<T>(
   try {
     return await fn(client);
   } on AuthenticationException catch (error) {
-    var message = '';
+    var hint = '';
+    var message = 'authentication failed';
 
+    assert(error.statusCode == 401 || error.statusCode == 403);
     if (error.statusCode == 401) {
       if (systemCache.tokenStore.removeCredential(hostedUrl)) {
         log.warning('Invalid token for $hostedUrl deleted.');
       }
-      message = '$hostedUrl package repository requested authentication! '
+      hint = '$hostedUrl package repository requested authentication!\n'
           'You can provide credential using:\n'
```

Comment:
```suggestion
          'You can provide credentials using:\n'
```

See https://english.stackexchange.com/a/252075